<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" >

<material name="white">
    <color rgba="1 1 1 1"/>
</material>

<material name="orange">
    <color rgba="1 0.3 0.1 1"/>
</material>

<material name="blue">
    <color rgba="0.2 0.2 1 1"/>
</material>

<material name="black">
    <color rgba="0 0 0 1"/>
</material>

<!-- Base link -->

<link name="base_link">
</link>


<!-- Servo Motor Pan Platform (Bottom) -->


<joint name="pan_platform_joint" type="continuous">
    <parent link="base_link"/>
    <child link="pan_platform"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
</joint>

<link name="pan_platform">
    <visual>
        <geometry>
            <cylinder length="0.04" radius="0.05" />
        </geometry>
        <material name="blue"/>
    </visual>
</link>

<!-- Servo Motor Tilt Platform (Bottom) -->

<joint name="tilt_platform_joint" type="revolute">
    <parent link="pan_platform"/>
    <child link="tilt_platform"/>

    <origin xyz="0 0 0.07" rpy="-${pi/2} 0 0"/>
    <axis xyz = "0 0 -1"/>
    <limit lower = "0" upper ="${pi}" velocity = "100" effort="100"/>
   
</joint>

<link name="tilt_platform">
    <visual>
        <geometry>
            <cylinder length="0.04" radius="0.05" />
        </geometry>
        <material name="orange"/>
    </visual>
</link>





</robot>
